src/core/cluster.py

```diff
@@ -394,13 +394,13 @@ def ready_to_start(self) -> Status:  # noqa: C901
     @property
     def _balancer_status(self) -> Status:
         """Checks for role=balancer specific readiness."""
-        if not self.runs_balancer:
+        if not self.runs_balancer or not self.unit_broker.unit.is_leader():
```

Comment:
**praise**: I think this is the missing piece to remove the flakiness in my own PR. THANK YOU